src/debugger/debugAdapterFactory.ts

```diff
@@ -17,19 +17,76 @@ import * as path from "path";
 import { WorkspaceContext } from "../WorkspaceContext";
 import { DebugAdapter, LaunchConfigType } from "./debugAdapter";
 import { Version } from "../utilities/version";
+import { registerLoggingDebugAdapterTracker } from "./logTracker";
+import { SwiftToolchain } from "../toolchain/toolchain";
+import { SwiftOutputChannel } from "../ui/SwiftOutputChannel";
 
-export function registerLLDBDebugAdapter(workspaceContext: WorkspaceContext): vscode.Disposable {
+/**
+ * Registers the active debugger with the extension, and reregisters it
+ * when the debugger settings change.
+ * @param workspaceContext  The workspace context
+ * @returns A disposable to be disposed when the extension is deactivated
+ */
+export function registerDebugger(workspaceContext: WorkspaceContext): vscode.Disposable {
+    let subscriptions: vscode.Disposable[] = [];
+    const register = async () => {
+        subscriptions.map(sub => sub.dispose());
+        subscriptions = [
+            registerLoggingDebugAdapterTracker(workspaceContext.toolchain.swiftVersion),
+            registerLLDBDebugAdapter(workspaceContext.toolchain, workspaceContext.outputChannel),
+        ];
+
+        await workspaceContext.setLLDBVersion();
+
+        // Verify that the adapter exists, but only after registration. This async method
+        // is basically an unstructured task so we don't want to run it before the adapter
+        // registration above as it could cause code executing immediately after register()
+        // to use the incorrect adapter.
+        DebugAdapter.verifyDebugAdapterExists(
+            workspaceContext.toolchain,
+            workspaceContext.outputChannel,
+            true
+        ).catch(error => {
+            workspaceContext.outputChannel.log(error);
+        });
+    };
+
+    const changeMonitor = vscode.workspace.onDidChangeConfiguration(event => {
```

Comment:
what is the change from the user perspective? This seems to do more than just add new tests, I guess mostly just enabling us to switch debuggers during a test run?

Reply:
This fixes a bug. Changing the debug adapter didn't require a restart of the extension. When you changed the debug adapter nothing re-registered the adapter/loggingDebugAdapter.

Reply:
It was uncovered by the tests since they did need to swift the debug adapter without restarting the extension

Reply:
> Changing the debug adapter didn't require a restart of the extension

do you mean changing _did_ require a restart?

Reply:
@award999 I mean it did require a restart, but there was no dialog that popped up prompting a restart.